news/_posts/2015-05-05-release-notes-2.12.0-M1.md

```diff
@@ -0,0 +1,163 @@
+Concretely, this means you may have external dependencies on your run-time classpath that use a different version of Scala than the one you're compiling with, as long as they're binary compatibile.
+In other words, separate compilation on different binary compatible versions does not introduce problems compared to compiling and running everything on the same version of Scala.
+
+We check binary compatibility automatically with [MiMa](https://github.com/typesafehub/migration-manager).
+
+#### Forwards and Back
+
+We distinguish forwards and backwards compatibility (think of these as properties of a sequence of versions, not of an individual version).
+Maintaining backwards compatibility means code compiled on an older version will link with code compiled with newer ones.
+Forwards compatibility allows you to compile on new versions and run on older ones.
+
+Thus, backwards compatibility precludes the removal of (non-private) methods, as older versions could call them, not knowing they would be removed, whereas forwards compatibility disallows adding new (non-private) methods, because newer programs may come to depend on them, which would prevent them from running on older versions (private methods are exempted here as well, as their definition and call sites must be in the same compilation unit).
+
+#### Meta
+
+Note that so far we've only talked about the jars generated by scalac for the standard library and reflection.
+Our policies do not extend to the meta-issue: ensuring binary compatibility for bytecode generated from identical sources, by different version of scalac?
+(The same problem exists for compiling on different JDKs.)
+While we strive to achieve this, it's not something we can test in general.
+Notable examples where we know meta-binary compatibility is hard to achieve: specialisation and the optimizer.
+
+In short, if binary compatibility of your library is important to you, use [MiMa](https://github.com/typesafehub/migration-manager) to verify compatibility before releasing.
+Compiling identical sources with different versions of the Scala compiler (or on different JVM versions!) could result in binary incompatible bytecode.
+This is rare, and we try to avoid it, but we can't guarantee it will never happen.
+
+#### Concretely
+
+We guarantee forwards and backwards compatibility of the `"org.scala-lang" % "scala-library" % "2.12.x"` and `"org.scala-lang" % "scala-reflect" % "2.12.x"` artifacts, except for anything under the `scala.reflect.internal` package, as scala-reflect is still experimental.
+We also strongly discourage relying on the stability of `scala.concurrent.impl` and `scala.reflect.runtime`, though we will only break compatibility for severe bugs here.
```

Comment:
This is misleading for milestones. We don't actually guarantee any kind of binary compatibility between milestones, do we?